core/templates/pages/exploration-editor-page/editor-tab/state-name-editor/state-name-editor.component.ts

```diff
@@ -102,7 +107,31 @@ angular.module('oppia').component('stateNameEditor', {
           RouterService.navigateToMainTab(normalizedStateName);
         }
       };
+
+      ctrl.windowOnload = function() {
+        ctrl.TabName = RouterService.getActiveTabName();
+        if (ctrl.TabName === 'main') {
+          FocusManagerService.setFocus('oppiaEditableSection');
+        }
+        if (ctrl.TabName === 'feedback') {
+          FocusManagerService.setFocus('newThreadButton');
+        }
+        if (ctrl.TabName === 'history') {
+          FocusManagerService.setFocus('usernameInputField');
+        }
+      };
+
       ctrl.$onInit = function() {
+        // If-user refreshes the tab.
+        $window.onload = ctrl.windowOnload();
+
+        $rootScope.$watch(
```

Comment:
Let's avoid using $watch. It isn't good for performance. Can we use event emitters instead?

Reply:
done